server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -144,21 +144,21 @@ protected TransportReplicationAction(Settings settings, String actionName, Trans
 
         this.transportPrimaryAction = actionName + "[p]";
         this.transportReplicaAction = actionName + "[r]";
-        registerRequestHandlers(actionName, transportService, request, replicaRequest, executor);
+
+        transportService.registerRequestHandler(actionName, request, ThreadPool.Names.SAME, this::handleOperationRequest);
+        transportService.registerRequestHandler(transportPrimaryAction, () -> new ConcreteShardRequest<>(request), executor, true,
+            forcePrimaryActionExecution(), this::handlePrimaryRequest);
```

Comment:
The result of `forcePrimaryActionExecution()` is passed to the wrong parameter (`canTripCircuitBreaker`). Swap this and the previous `true` and I think it will work.

Reply:
Thanks, well spotted. I opened 40762.